### PR TITLE
Improve error messages when base pipe operator is used wrongly

### LINKED ISSUE #4792: suggested tweak to the error message in validate_mapping()
It's been *very* helpful to trap the use of pipe rather than `+`.
I'm not sure if it's possible to tell whether the base pipe operator is used instead of the magrittr version.

Perhaps it might be better to say:

`Did you use %>% or |> instead of +?`

``` r
suppressWarnings(library(tidyverse))
mtcars |>
  ggplot(aes(x = mpg)) |> geom_histogram(bins = 10)
#> Error in `validate_mapping()`:
#> ! `mapping` must be created by `aes()`
#> Did you use %>% instead of +?
```

<sup>Created on 2022-04-01 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Thanks as always for your efforts.

### PR DESCRIPTION
**Pull request by 92amartins**
This PR is an attempt to fix #4792.

I tried to follow the @yutannihilation guidelines on the original issue.

Couldn't figure it out how to include proper snaps.

Further guidance is greatly appreciated

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # ggplot2 (development version)
 
+* improved error messages when `|>` is used instead of `+` (@92amartins, #4792)
```
Comment (yutannihilation) [inline NEWS.md:3]:
```suggestion
* Improved error messages when `|>` is used instead of `+` (@92amartins, #4792)
```

**Review by yutannihilation — APPROVED** [review]
Looks good, thanks!

**Comment by yutannihilation** [comment]
The CI failure is a temporary one due to the release of R 4.2. We can ignore it for now.

**Comment by 92amartins** [comment]
@yutannihilation no problem! I'll search for other good issues to contribute (using the `good first issue` and `help wanted` labels). Let me know if you have any advice for my first contribution.

**Comment by yutannihilation** [comment]
The ones with "good first issue" label should be good for the first contribution!

https://github.com/tidyverse/ggplot2/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue+%3Aheart%3A%22

**Pull request closed without merging**